Source/FieldSolver/FiniteDifferenceSolver/MacroscopicEvolveE.cpp

```diff
@@ -145,9 +173,17 @@ void FiniteDifferenceSolver::MacroscopicEvolveECartesian (
                                                 i, j, k, amrex::IntVect(1,0,0) );
                 amrex::Real mu = T_MacroAlgo::macro_avg_to_edge(i, j, k, amrex::IntVect(2,1,1),
                                                     mu_arr);
-                Ex(i, j, k) = alpha * Ex(i, j, k) + (beta/mu)
+#ifdef WARPX_MAG_LLG
+                Ex(i, j, k) = alpha * Ex(i, j, k) + beta
+                     * ((- T_Algo::DownwardDz(By, coefs_z, n_coefs_z, i, j, k, 0)
+                         + T_Algo::DownwardDy(Bz, coefs_y, n_coefs_y, i, j, k, 0))/PhysConst::mu0
+                 - T_Algo::DownwardDz(M_yface, coefs_z, n_coefs_z, i, j, k, 1)
+                         + T_Algo::DownwardDy(M_zface, coefs_y, n_coefs_y, i, j, k, 2));
+#else
```

Comment:
overarching comment: if H_maxwell = B/mu_0 - M, then shouldn't the signs on all the M_ terms be reversed?

Reply:
Done!